### PR TITLE
stdlib: Add [CommandLine] interface to get command line args

### PR DESCRIPTION
**Pull request by vtourneur**
In order to get command line arguments from the Coq side, the program
must be run with the `fsrun` script (in the `run` directory). It uses
environment variables to transmit the args given by the user. It is also
no longer necessary to add `Exec xxx.` at the end of the program file,
since `fsrun` will append `Exec main.` to the file when calling `coqc`.
`fsrun` will also remove the first line of the program file if it begins
with `#!`, so it is possible to add `#!/usr/bin/env fsrun` and run the
program with `./Prog.v`, if `fsrun` is accesible in the environment path.

### CODE REVIEW

**Comment by lthms** [comment]
Awesome PR @vtourneur. There are a lot of very good ideas in this PR. I have several remarks, though. Feel free to ask any questions you might have. I personally think it could be a good idea that we have a discussion about this.

I don’t think `fsrun` will be usable as is, for at least two reasons:

1. You cannot use Proof General anymore in Emacs
2. I don’t think `fsrun` in its current state handles programs implemented across several files

I think we should take a slightly different approach, where `fsrun` would work like that

```
fsrun <COQ FILE> [-- <ARGS>+] 
```

Internally, it would generate a Coq file in `/tmp` of the form

```coq
Require Import FreeSpec.Exec.
Add LoadPath "<ABSOLUTE PATH TO <COQ FILE> DIRECTORY>".
Require Import <COQ FILE MODULE>.

Exec main.
```

Using `coqc` on this file should do the trick.

This approach requires the coq file (and its dependencies) to be compiled (maybe `fsrun` could check that?), but it should really improve start-up time and that is very interesting.

What do you think?

Regarding the `CommandLine` interface, I like the idea of using environment variables, but I would have taken a different approach to achieve the same goal: creating a generic interface to get and set environment variable value, and then define Coq programs that use this interface to deal with command line arguments.

Note that I have in my pipe a couple of functions I want to add to FreeSpec.Core so you could declare and use your `CommandLine` interface (although I would have called it `CommandArgs`), but implement it using pure Gallina terms and using the environment variables interface.

To summarize: your PR objective is very nice, and I agree that using environment variables is a good idea to deal with command-line arguments right now. I really think you should discuss further in order to improve its implementation, though.

(also, I am not fond of the name—but I have to admit I don’t have any clever suggestion to bring on the table! maybe `coqr` or `coqrun`)

**Comment by vtourneur** [comment]
This commit fixes the two issues you mentioned, but this is now a little bit more complicated to use:
* The file(s) containing the program have to be compiled with `coqc` (one of these files must contain the `main` program).
* A new script `fsc` has to be called with all the `.vo` generated.
* It will create an executable file which can be run with `fsrun`.

**Comment by lthms** [comment]
I will close this for now, as (1) #23 reuse your proposal about how to deal with CLI arguments, and (2) we have agreed to keep tools like `fsrun` outside of the FreeSpec repo for now.

Thanks a lot for the original design!
